lib/client.js

```diff
@@ -126,6 +129,23 @@ function messageHandler(client, event) {
   }
 }
 
+// When doing singular operations and we retrieve and error this function will throw that error.
+function processSingleRequest(path, result) {
```

Comment:
This seems to process the response, not the request and it's not processing single requests only, can we rename it to `processResponse`?